src/hepqpr/qallse/data_structures.py

```diff
@@ -236,13 +236,44 @@ def doublets(self) -> List[Doublet]:
         #: Does this need to be reworked with the slicing algorithm?
         return self.t1.doublets() + [self.t2.d2]
 
-
 class QuboSlice(object):
 # TODO Make QUBO slice class
     def __init__(self, q: {}, eta: int, phi: int):
-
-        self.qubo: {} = q
+        self.qubo = q
         self.eta = eta
         self.phi = phi
+    
+    def __str__(self):
+        return 'eta: {0}, phi: {1}\nQubo: {2}'.format(self.eta,self.phi,self.qubo)
+
+#container carrying all qubo slices
+class SliceContainer(object):
```

Comment:
This class is a container for all the qubo-slices you create. It will help you with bookkeeping!